newm-server/src/main/kotlin/io/newm/server/features/distribution/eveara/EvearaDistributionRepositoryImpl.kt

```diff
@@ -0,0 +1,136 @@
+                    contentType(ContentType.Application.Json)
+                    accept(ContentType.Application.Json)
+                    setBody(
+                        EvaraApiGetTokenRequest(
+                            clientId = evearaClientId,
+                            clientSecret = evearaClientSecret,
+                        )
+                    )
+                }
+                if (!response.status.isSuccess()) {
+                    throw ServerResponseException(response, "Error getting Eveara accessToken!")
+                }
+                val apiTokenResponse: EvearaApiTokenResponse = response.body()
+                if (!apiTokenResponse.success) {
+                    throw ServerResponseException(response, "Error getting Eveara accessToken! success==false")
+                }
+                val apiToken = ApiToken(
+                    accessToken = apiTokenResponse.accessToken,
+                    refreshToken = apiTokenResponse.refreshToken,
+                    expiryTimestampMillis = System.currentTimeMillis() + apiTokenResponse.expiresInSeconds * 1000L
+                )
+                apiTokenCache.put(-1, apiToken)
+
+                apiToken.accessToken
+            }
+        }
+    }
+
+    override suspend fun getGenres(): GetGenresResponse {
+        val response = httpClient.get("https://$evaraServer/api/v2.0/genres") {
+            accept(ContentType.Application.Json)
+            bearerAuth(getEvearaApiToken())
+        }
+        if (!response.status.isSuccess()) {
+            throw ServerResponseException(response, "Error getting genres!")
+        }
+        val getGenresResponse: GetGenresResponse = response.body()
+        if (!getGenresResponse.success) {
+            throw ServerResponseException(response, "Error getting genres! success==false")
+        }
+
+        return getGenresResponse
+    }
+
+    override suspend fun distributeSong(song: SongEntity) {
+        // TODO: Implement later. Focus on completing minting code first. Assume we have distributed to Eveara successfully.
+    }
+
+    private data class EvaraApiGetTokenRequest(
+        @SerialName("grant_type")
+        val grantType: String = "client_credentials",
+        @SerialName("client_id")
+        val clientId: String,
+        @SerialName("client_secret")
+        val clientSecret: String,
+    )
+
+    @Serializable
+    private data class EvearaApiTokenResponse(
+        @SerialName("success")
+        val success: Boolean,
+        @SerialName("access_token")
+        val accessToken: String,
+        @SerialName("refresh_token")
+        val refreshToken: String,
+        @SerialName("expires_in")
+        val expiresInSeconds: Long,
+    )
+
+    private data class ApiToken(
+        val accessToken: String,
+        val refreshToken: String,
+        val expiryTimestampMillis: Long,
+    )
+
+    companion object {
+        private const val FIVE_MINUTES_IN_MILLIS = 300_000L
```

Comment:
We should probably move this value to the .conf file. I can do it later when I do the other work.